includes/payments/class-payments-query.php

```diff
@@ -242,38 +249,97 @@ public function get_payments() {
 	}
 
 	/**
-	 * If querying a specific date, add the proper filters.
+	 * Get payments by group
 	 *
-	 * @since  1.0
+	 * @since  1.8.17
 	 * @access public
 	 *
-	 * @return void
+	 * @return array
 	 */
-	public function date_filter_pre() {
-		if ( ! ( $this->args['start_date'] || $this->args['end_date'] ) ) {
-			return;
+	public function get_payment_by_group() {
+		global $wpdb;
+
+		$allowed_groups = array( 'post_status' );
+		$result         = array();
+
+
+		if ( in_array( $this->args['group_by'], $allowed_groups ) ) {
+			// Set only count in result.
+			if ( $this->args['count'] ) {
+
+				$this->set_filters();
+
+				$new_results = $wpdb->get_results( $this->get_sql(), ARRAY_N );
+
+				$this->unset_filters();
+
+				foreach ( $new_results as $results ) {
+					$result[ $results[0] ] = $results[1];
+				}
+
+				switch ( $this->args['group_by'] ) {
+					case 'post_status':
+
+						/* @var Give_Payment $donation */
+						foreach ( give_get_payment_status_keys() as $status ) {
+							if ( ! isset( $result[ $status ] ) ) {
+								$result[ $status ] = 0;
+							}
+						}
+
+						break;
+				}
+			} else {
+				$donations = $this->get_payments();
+
+				/* @var $donation Give_Payment */
+				foreach ( $donations as $donation ) {
+					$result[ $donation->{$this->args['group_by']} ][] = $donation;
+				}
+			}
 		}
 
-		$this->setup_dates( $this->args['start_date'], $this->args['end_date'] );
 
-		add_filter( 'posts_where', array( $this, 'payments_where' ) );
+		/**
+		 * Filter the result
+		 *
+		 * @since 1.8.17
+		 */
+		return apply_filters( 'give_get_payment_by_group', $result, $this );
 	}
 
 	/**
-	 * If querying a specific date, remove filters after the query has been run
-	 * to avoid affecting future queries.
+	 * If querying a specific date, add the proper filters.
 	 *
 	 * @since  1.0
 	 * @access public
 	 *
 	 * @return void
 	 */
-	public function date_filter_post() {
+	public function date_filter_pre() {
 		if ( ! ( $this->args['start_date'] || $this->args['end_date'] ) ) {
 			return;
 		}
 
-		remove_filter( 'posts_where', array( $this, 'payments_where' ) );
+		$this->setup_dates( $this->args['start_date'], $this->args['end_date'] );
```

Comment:
@DevinWalker Instead of using custom filter for date query we are using `date_query` param of `WP_Query`